src/material-experimental/mdc-checkbox/checkbox.scss

```diff
@@ -19,6 +20,10 @@
 }
 
 .mat-mdc-checkbox {
+  // MDC theme styles also include structural styles so we have to include the theme at least
+  // once here. The values will be overwritten by our own theme file afterwards.
+  @include mdc-checkbox-theme.theme-styles(mdc-checkbox-theme.$light-theme);
```

Comment:
Is there a plan to eventually not have to repeat the styles twice?

Reply:
There's a whole [discussion about it here](https://github.com/angular/components/pull/23143#discussion_r669035932), but the tl;dr is that it's working as expected. The `theme-styles` include the "slots" which will consume the CSS variables. Then in the actual `theme` mixins only the variable values are overwritten.

Reply:
These styles should be placed on the inner `mdc-checkbox` element rather than the host. There's some structural styles being applied that assume its being placed on the checkbox itself (check out the demo app to see that the checkbox have much larger height/width).

Reply:
Fixed.